File: TH_Julia/Bubble_Sort_Bai2.jl
# Bubble Sort function
function bubble_sort!(arr)
    n = length(arr)
    for i in 1:n-1
        swapped = false
        for j in 1:n-i
            if arr[j] > arr[j + 1]
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = true
            end
        end
        if !swapped
            break
        end
    end
end

# Main function
function main()
    arr = [5, 2, 9, 1, 5]
    bubble_sort!(arr)
    println("Ket qua sau Bubble Sort: ", arr)
end

main()

File: TH_Buoi2/Bubble_Sort_Bai2.cpp
#include <iostream>
#include <vector>
using namespace std;

// Hàm Bubble Sort
void BubbleSort(vector<int> &arr)
{
    int n = arr.size();
    for (int i = 0; i < n - 1; i++)
    {
        bool swapped = false;
        for (int j = 0; j < n - i - 1; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                swap(arr[j], arr[j + 1]);
                swapped = true;
            }
        }
        if (!swapped)
            break;
    }
}  
int main()
{
    vector<int> arr = {5, 2, 9, 1, 5};
    BubbleSort(arr);
    cout << "Ket qua sau Merge Sort: ";
    for (int x : arr)
    cout << x << " ";
    cout << endl;
    return 0;
}